pkg/scheduler/plugins/task-topology/topology.go

```diff
@@ -247,10 +247,15 @@ func affinityCheck(job *api.JobInfo, affinity [][]string) error {
 
 	var taskNumber = len(job.Tasks)
 	var taskRef = make(map[string]bool, taskNumber)
+	var jobNamePrefix = job.Name + "-"
 	for _, task := range job.Tasks {
-		tmpStrings := strings.Split(task.Name, "-")
-		if _, exist := taskRef[tmpStrings[len(tmpStrings)-2]]; !exist {
-			taskRef[tmpStrings[len(tmpStrings)-2]] = true
+		// the full task name looks like "${job name}-${task name}-${index}",
+		// so we can trim the jobNamePrefix and the indexSuffix to get the short task name.
+		tmpTaskName := task.Name[:strings.LastIndex(task.Name, "-")]
+		tmpTaskName = strings.TrimPrefix(tmpTaskName, jobNamePrefix)
```

Comment:
Rename tmpTaskName to taskName is better.

Reply:
I'm concerned that this might lead to ambiguity, as tmpTaskName is not the actual name of the task.

Reply:
I did not notice that, you're right: )